### ISSUE TITLE
Overzealous "Lint/SafeNavigationChain" cop

### ISSUE THREAD

**Issue by agalloch** [bbatsov/rubocop#3957]
The `Lint/SafeNavigationChain` cop tries to do a little too much when safeguarding for `nil`s.

## Expected behavior
```ruby
# Whatever the `foo`, calling `to_s` on it should be safe, 
# so no need for `&` operator
foo.to_s.tr('.', ',')
```
## Actual behavior
```ruby
# Cop requires this form which is arguably better/safer
foo&.to_s&.tr('.', ',')
```
Perhaps it could be configured to whitelist/ignore specific methods to avoid extra noise in method chains?

Thank you!

**Comment by mikegee** [comment]
I'm having trouble reproducing this issue. Rubocop doesn't complain about `foo.to_s.tr('.', ',')` on master with the default configuration. Perhaps you have an older Rubocop or some non-default configuration?

**Comment by agalloch** [comment]
Yes, it turned out to be a false positive when using lonely operator in a silly way: `foo&.to_s.tr('.', ',')`.

Thank you for the swift response, closing the issue :)
